public/app/plugins/datasource/jaeger/datasource.ts

```diff
@@ -48,14 +48,25 @@ export class JaegerDatasource extends DataSourceApi<JaegerQuery, JaegerJsonData>
     return res.data.data;
   }
 
+  // Return false if the search form has not been filled in sufficiently. Currently, this only checks if a service
+  // has been selected, but this func can later be expanded if the search form becomes more complex.
+  isFormValid(query: JaegerQuery): boolean {
```

Comment:
Comments are a great tool but we must use it sparingly and reserve it for cases where they will make a difference to the person reading the code. Always remember that when you add a comment you're also asking all developers that will change this piece of code to also make sure they update the comment with the respective changes, increasing their work load or, even worse, having the comment fall out of sync from the code it's referencing.

In this case it would be better to not have the comment and try to clarify the code in other ways, for example, by renaming the function to `isSearchFormValid(...)`, which should be enough to provide the context for the function.

Reply:
Good point. I've removed the comment and changed the name of the function to `isSearchFormValid` to be more specific about what the function does. See commit https://github.com/grafana/grafana/pull/54172/commits/fbeb8c5cb63af10e825591da5d639e6360eec1c1.

Reply:
Good point. I've removed the comment and changed the name of the function to `isSearchFormValid` to be more specific about what the function does. See commit https://github.com/grafana/grafana/commit/fbeb8c5cb63af10e825591da5d639e6360eec1c1.